static/app/utils/useSyncedLocalStorageState.tsx

```diff
@@ -0,0 +1,49 @@
+import {useCallback, useEffect} from 'react';
+
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+type SyncedLocalStorageEvent = CustomEvent<{key: string; value: any}>;
+
+const SET_VALUE_EVENT_NAME = 'synced-local-storage';
+
+/**
+ * Same as `useLocalStorageState`, but notifies and reacts to state changes.
+ * Use this you want to access local storage state from multiple components
+ * on the same page.
+ */
+export function useSyncedLocalStorageState<S>(
+  key: string,
+  initialState: S | ((value?: unknown, rawValue?: unknown) => S)
+): [S, (value: S) => void] {
+  const [value, setValue] = useLocalStorageState(key, initialState);
+
+  const setValueAndNotify = useCallback(
```

Comment:
👏🏼 👏🏼 for sync dispatching this inside a callback - I sometimes see dispatches happen inside useEffect when value changes and that is almost always prone to bugs or values flickering in the UI.